### PR TITLE
Option to build oap with spark souce code which applid numa code

### PR DESCRIPTION
**Pull request by yeyuqiang**
## What changes were proposed in this pull request?

An option script for the user to build OAP with numa code and no need for them to change the spark source code. Support binding numa to executors in balance and fault-tolerant.

## How was this patch tested?
Start two executors and kill one several times to check whether the restarted executor will bind numa in balance.
numa

### CODE REVIEW

**Comment by yeyuqiang** [comment]
@zhixingheyi-tian @winningsix  Noticed that you've contribute a spark numa patch. but do we also need a patch for SparkEnv. I failed to test the current numa patch. So I cherry pick the code from branch OPEN_SOURCE_VMEMCACHE_SPARK_2.4.4_PREVIEW. Please help to review.

File: pom.xml [hunk 1]
```diff
@@ -807,5 +807,71 @@
...
+            <version>1.8</version>
+            <executions>
+              <execution>
+                <phase>generate-sources</phase>
+                <goals>
+                  <goal>run</goal>
+                </goals>
+                <configuration>
+                  <tasks>
+                    <echo>Copy numa pacthed file as OAP source code</echo>
```
Comment (winningsix) [inline pom.xml]: Support ````NUMA``` in ``Spark```

File: scripts/apply_patch_to_oap.sh [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    if [ -z $mount_point ]; then
+      echo "Please mount /dev/$pmem_device at the initial path which numa node id is $i in persistent-memory.xml"
+    else
+      echo "Current /dev/$pmem_device mounted at $mount_point."
+      echo "Please check if $mount_point is associate with numa node id $i in persistent-memory.xml."
+    fi
+  done
+}
+
+patch_to_oap() {
```
Comment (winningsix) [inline scripts/apply_patch_to_oap.sh]: Hint: method name to ```apply_numa_patch_oap```

File: scripts/apply_patch_to_spark.sh [hunk 3]
```diff
@@ -0,0 +1,128 @@
+#!/bin/bash
+
+SPARK_UPSTREAM_URL="https://github.com/apache/spark/archive/"
+SPARK_VERSION_SUPPORTED=(
```
Comment (winningsix) [inline scripts/apply_patch_to_spark.sh]: I am thinking whether we need to do it. Typically, OAP is bumping to specific Spark version. We should update the patch file (if needed) when supporting new Spark versions.

File: scripts/apply_patch_to_spark.sh [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+  SPARK_SOURCE_DIR=$DOWNLOADED_SPARK_DIR/spark_source/$SPARK_SOURCE_NAME
+  if [ -d $SPARK_SOURCE_DIR ]; then
+    rm -rf $SPARK_SOURCE_DIR
+  fi
+  tar -zxf $TARBALL_NAME -C $DOWNLOADED_SPARK_DIR/spark_source
+  rm $TARBALL_NAME
+}
+
+patch_to_spark() {
+  # TODO: we'd better use /usr/bin/patch to apply patch. Here seems the current patch file is not applicable
```
Comment (winningsix) [inline scripts/apply_patch_to_spark.sh]: How did you create your patch? Which command?

**Comment by knightyyq** [comment]
Since the spark with numa patch is already provided. I'll close it first.

**Pull request closed without merging**